hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -300,7 +300,7 @@ && equal( join.getKey(), manyToOne ) ) {
 				}
 			}
 			isOptional = ( (ManyToOne) bootValue ).isIgnoreNotFound();
-			isInternalLoadNullable = ( isNullable && bootValue.isForeignKeyEnabled() ) || notFoundAction == NotFoundAction.IGNORE;
+			isInternalLoadNullable = ( isNullable && bootValue.isForeignKeyEnabled() ) || hasNotFoundAction();
```

Comment:
Of course: if the not found action is EXCEPTION, the initializers (e.g. [EntitySelectFetchInitializer](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializer.java#L194-L205)) expect that a `null` entity instance to be retrieved, to then manually throw a `FetchNotFoundException` in that case.

This change was needed because if `isInternalLoadNullable` was false the `interalLoad` method would throw a different error instead.